lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+                                is_default_cache=False):
+        try:
+            if is_default_cache and cache_behavior is None:
+                cache_behavior = {}
+            if cache_behavior is None and valid_origins is not None:
+                return None
+            if 'min_ttl' not in cache_behavior:
+                cache_behavior['min_t_t_l'] = (
+                    self.__default_cache_behavior_min_ttl)
+            else:
+                cache_behavior = helpers.change_dict_key_name(
+                    cache_behavior, 'min_ttl', 'min_t_t_l')
+            if 'max_ttl' not in cache_behavior:
+                cache_behavior['max_t_t_l'] = (
+                    self.__default_cache_behavior_max_ttl)
+            else:
+                cache_behavior = helpers.change_dict_key_name(
+                    cache_behavior, 'max_ttl', 'max_t_t_l')
+            if 'default_ttl' not in cache_behavior:
+                cache_behavior['default_t_t_l'] = (
+                    self.__default_cache_behavior_default_ttl)
+            else:
+                cache_behavior = helpers.change_dict_key_name(
+                    cache_behavior, 'default_ttl', 'default_t_t_l')
+            if 'compress' not in cache_behavior:
+                cache_behavior['compress'] = (
+                    self.__default_cache_behavior_compress)
+            if is_default_cache:
+                if 'target_origin_id' not in cache_behavior:
+                    cache_behavior['target_origin_id'] = (
+                        self.get_first_origin_id_for_default_cache_behavior(
+                            valid_origins))
+            cache_behavior = self.validate_forwarded_values(
+                cache_behavior.get('forwarded_values'), cache_behavior)
+            cache_behavior = self.validate_allowed_methods(
+                cache_behavior.get('allowed_methods'), cache_behavior)
+            cache_behavior = self.validate_lambda_function_associations(
+                cache_behavior.get('lambda_function_associations'),
+                cache_behavior)
+            if 'viewer_protocol_policy' not in cache_behavior:
+                cache_behavior['viewer_protocol_policy'] = (
+                    self.__default_cache_behavior_viewer_protocol_policy)
+            else:
+                self.validate_attribute_with_allowed_values(cache_behavior.get(
+                    'viewer_protocol_policy'),
+                    'cache_behavior.viewer_protocol_policy',
+                    self.__valid_viewer_protocol_policies)
+            if 'smooth_streaming' not in cache_behavior:
+                cache_behavior['smooth_streaming'] = (
+                    self.__default_cache_behavior_smooth_streaming)
+            cache_behavior['trusted_signers'] = self.validate_trusted_signers(
+                cache_behavior.get('trusted_signers'))
+            return cache_behavior
+        except Exception as e:
```

Comment:
refactored this function into another function and removed the try/except block so all errors are caught further down now. let me know if this is not what you meant.